.github/workflows/firebase_actions.yml

```
@@ -25,10 +25,27 @@ jobs:
       - name: Install Firebase CLI
         run: curl -sL https://firebase.tools | bash
 
-      - name: NPM install
+      - name: Install NPM dependencies to the firebase folder
         run: npm install
         working-directory: metrics/firebase/
 
+      - name: Install NPM dependencies to the functions folder
+        run: npm install
+        working-directory: metrics/firebase/functions/
+
+      - uses: cedx/setup-dart@v2
+        with:
+          release-channel: beta
+          version: 2.12.0-133.2.beta
+
+      - name: Get dependencies
+        run: pub get
+        working-directory: metrics/firebase/functions/
+
+      - name: Run functions tests
```

```suggestion
      - name: Run Cloud Functions tests
```
Also, would it be great to run the `dartanalyzer` before running tests?